src/modules/App/hooks/useApplyTextDirection.ts

```diff
@@ -22,14 +23,15 @@ const ELEMENT_IDS = [
  * attribute set explicitly to ensure proper directionality based on the app's language setting.
  */
 const useApplyTextDirection = (direction: HTMLTextDirection) => {
+  const { isMobile } = useMediaQueryContext();
   useEffect(() => {
     ELEMENT_IDS.forEach((id) => {
       const element = document.getElementById(id);
       if (element) {
         element.dir = direction;
       }
     });
-  }, [direction]);
+  }, [direction, isMobile]);
```

Comment:
`isMobile` is determined within `useMediaQueryContext`, and this needs to run separately of the `useApplyTextDirection` hook. This ensures that the internal logic of the `useApplyTextDirection` hook is executed only after the `isMobile` value has been properly set.